Analysis/EventFiltering/centralEventFilterProcessor.cxx

```diff
@@ -0,0 +1,185 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// @file   centralEventFilterProcessor.cxx
+
+#include "centralEventFilterProcessor.h"
+
+#include "Framework/ControlService.h"
+#include "Framework/ConfigParamRegistry.h"
+#include "Framework/TableConsumer.h"
+
+#include <TFile.h>
+#include <TH1D.h>
+
+#include <iostream>
+#include <cstdio>
+#include <fmt/format.h>
+#include <rapidjson/document.h>
+#include <rapidjson/filereadstream.h>
+
+using namespace o2::framework;
+using namespace rapidjson;
+
+namespace
+{
+Document readJsonFile(std::string& config)
+{
+  FILE* fp = fopen(config.data(), "rb");
+
+  char readBuffer[65536];
+  FileReadStream is(fp, readBuffer, sizeof(readBuffer));
+
+  Document d;
+  d.ParseStream(is);
+  fclose(fp);
+  return d;
+}
+} // namespace
+
+namespace o2::aod::filtering
+{
+
+void CentralEventFilterProcessor::init(framework::InitContext& ic)
+{
+  // JSON example
+  // {
+  //   "subwagon_name" : "CentralEventFilterProcessor",
+  //   "configuration" : {
+  //     "NucleiFilters" : {
+  //       "H2" : 0.1,
+  //       "H3" : 0.3,
+  //       "HE3" : 1.,
+  //       "HE4" : 1.
+  //     }
+  //   }
+  // }
+  std::cout << "Start init" << std::endl;
+  Document d = readJsonFile(mConfigFile);
+  int nCols{0};
+  for (auto& workflow : d["workflows"].GetArray()) {
+    if (std::string_view(workflow["subwagon_name"].GetString()) == "CentralEventFilterProcessor") {
+      auto& config = workflow["configuration"];
+      for (auto& filter : AvailableFilters) {
+        auto& filterConfig = config[filter];
+        if (config.IsObject()) {
+          std::unordered_map<std::string, float> tableMap;
+          for (auto& node : filterConfig.GetObject()) {
+            tableMap[node.name.GetString()] = node.value.GetDouble();
+            nCols++;
+          }
+          mDownscaling[filter] = tableMap;
+        }
+      }
+      break;
+    }
+  }
+  std::cout << "Middle init" << std::endl;
+  mScalers = new TH1D("mScalers", ";;Number of events", nCols + 1, -0.5, 0.5 + nCols);
```

Comment:
Why not using the `HistogramRegistry`?

Reply:
We looked into it but we found easier and faster in our case (2 histogram written in a separate file) to do it like this. Are there some cool stuff/optimisations we are missing by not using it?

Reply:
With just 2 histograms used, i think that adding the histogram registry on top is an overhead.